### PR TITLE
Add support for -m and -f cli flags 

### PR DESCRIPTION
**Pull request by alubbe**
Closes https://github.com/AndersDJohnson/use-yarn/issues/1

I've done some minor refactoring and I kept your pre-ES6 style for backwards compatibility.

Let me know if you need anything else to approve this PR.

### CODE REVIEW

**Comment by alubbe** [comment]
ping @AndersDJohnson 
Do you need an update to the README?

**Comment by AndersDJohnson** [comment]
@alubbe Thanks for the contribution! Yes, if you could update the README in this PR, that would be great.

File: cli.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+var argv = require('minimist')(process.argv.slice(2));
+var useYarn = require('.')
 
-useYarn()
+var message
+if (argv.f) {
+  message = fs.readFileSync(argv.f, 'utf8')
+}
+
+if (!message && argv.m) {
```
Comment (AndersDJohnson) [inline cli.js]: Any reason for this instead or `else if(argv.m)`?

File: cli.js [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+  message = argv.m
+}
+
+if (!message) {
```
Comment (AndersDJohnson) [inline cli.js]: ...and `else`?

File: cli.js [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+  message = fs.readFileSync(path.resolve(__dirname, 'message.txt'), 'utf8')
+}
+
+useYarn(message)
```
Comment (AndersDJohnson) [inline cli.js]: Do you think we should accept file paths and support the default message with the programmatic API?

File: index.js [hunk 4]
```diff
@@ -1,10 +1,7 @@
 var isNpmNotYarn = require('is-npm-not-yarn')
-var fs = require('fs')
-var path = require('path')
 
 function useYarn(message) {
   if (isNpmNotYarn && process.env.DISABLE_USE_YARN !== 'true') {
-    message = message || fs.readFileSync(path.resolve(__dirname, 'message.txt'), 'utf8')
```
Comment (AndersDJohnson) [inline index.js]: I don't think we should remove support for the default message with the programmatic API, that's a breaking change.

**Review by AndersDJohnson — CHANGES_REQUESTED** [review]
Just a few comments.

File: cli.js (hunk 1 above)
Comment (alubbe) [reply cli.js]: Nope, that's much better :)

File: cli.js (hunk 3 above)
Comment (alubbe) [reply cli.js]: We probably need to change the function signature to allow this, but I have no use case for it. The consumer could also read the file themselves and then just pass on the message.

File: index.js (hunk 4 above)
Comment (alubbe) [reply index.js]: Will restore it!

**Comment by alubbe** [comment]
Okay, I have added some line to the README (feel free to improve them) and have restored the default message in the programmatic api. Any other input? @AndersDJohnson

File: README.md [hunk 5]
```diff
@@ -35,6 +35,24 @@ Or if you're on `npm >=5` or have [`npx`][npx]:
...
+You may provide a custom message via the `-m` flag:
+```
+{
+  "scripts": {
+     "preinstall": "npx use-yarn -m 'Please use yarn!'"
+  }
+}
+```
+
+Or, you may also provide a custom message read from a file via the `-f` flage:
```
Comment (AndersDJohnson) [inline README.md]: Typo "flage" should be "flag".

**Review by AndersDJohnson — DISMISSED** [review]
Just one more typo, we're close!

**Comment by alubbe** [comment]
Done

**Comment by alubbe** [comment]
This'll be a nice addition, looking forward to seeing this on npm :)

**Comment by AndersDJohnson** [comment]
@alubbe Published to npm with `2.2.0`. Thanks again!

**Pull request merged**